### DISCUSSION TITLE
google golang  :=   :=   :=   ???when  comes ??

### DISCUSSION THREAD

**Discussion by icetech233** [dotnet/csharplang#1594] [Language Ideas]
:= syntax as an alternative to var

like  

 a := 100

b :=  new xxxclasss()

c :=  "string"

**Comment by HaloFour** [comment]
Relevant:

https://github.com/dotnet/roslyn/issues/16442

**Comment by alrz** [comment]
I have an opinion about golang in general but I tend to keep it to myself for the sake of this repo.

**Comment by DavidArno** [comment]
Seriously, why all the negativity? This saves us two characters per assignment. This is huge! And it gives us three ways of declaring a variable and assigning it, which is clearly better than only the two ways we have now.

😇

**Comment by bbarry** [comment]
Two?

```csharp
int a = 1;
var b = 2;
void Declare(int x, out int y) => y = x;
Declare(3, out int c);
Declare(4, out var d);
if(5 is int e) { ... }
if(6 is var f) { ... }
var (g, _, _) = (7, 0, 0); // not sure if I have tuple deconstruction right here, but that is the syntax I am going for with some discards for good measure
```

Did I miss any?

**Comment by yaakov-h** [comment]
@bbarry switch/case and catch?

**Comment by AustinBryan** [comment]
Talk about lazy 😂

**Comment by fanoI** [comment]
Why change the sintax of C# to be to one of Go? It will make a lot more sense to do a sort of Go.Net working on the CLR....

**Comment by icetech233** [comment]
@bbarry what you means？

**Comment by CyrusNajmabadi** [comment]
I'm a person that really likes the shorthand form that go provides.  However, one thing i really dislike is that go doens't actually consistently let you use it.  At the top level, for example, i can't write:

```go
x := 1
```

I have to write:

```go
var x = 1
```

However, inside a function i *can* write ```x := 1```.  Ostensibly i've head this is to make top level parsing easier (as you need a keyword to start).  It rings a little hollow to me though as ```id :=``` is just a single token of lookahead.  Also, if it aids parsing... i'm not sure why it's hten ok to not require the same inside a function.

All-in-all, it's a good feature given the *tons* of variables you will have in code.  But it's also sucky because it's not something you can use consistently.  IMO of course :)

**Comment by The-Futurist** [comment]
@alrz 

> I have an opinion about golang in general but I tend to keep it to myself for the sake of this repo.

I do too, it's [not a very good language](http://yager.io/programming/go.html) and having explored this recently I was impressed at the huge difference in design quality of the language Rust.
